# DATA SCIENCE ANALYSIS WITH PYTHON
# Midterm take-home

### Due 2/19/2018 11:59 pm

<p>Instructions
<ol>
<li>Download the files <i>medical_payment.csv</i> and <i>geo-data.csv</i> into the same folder
<li>Read the file <i>Medical_Payment-pdf.pdf</i>
<li>Run the code below and then answer the questions
</ol></p>

<b>Penalties:</b> You will incur penalties if:
<ul>
<li>Your answer is different from the correct one</li>
<li>Your code is unnecessarily slow</li>
<li>Your code is longer than specified</li>
<li>You will be penalized if, in an attempt to limit the lines of code, you make your code too hard to read or too slow -- for example, by copy-pasting pieces of code in the same line instead of declaring a variable in one line and using the variable in another.
</ul>

In [1]:
import pandas as pd
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"
import warnings
warnings.filterwarnings('ignore')

In [2]:
df = pd.read_csv("medical_payment.csv",dtype=str).\
        rename(columns=lambda x: x.strip())

## Data Description

With this data set, people can see the price difference for a hundred common illnesses and procedures. Hospitals can set their own prices and differ substantially in some cases.
 
Before you move on, read the <b>Medical_Payment-pdf.pdf</b> to understand the dataset.

In [3]:
df.shape

(163065, 10)

The columns are as follows:

In [4]:
df.columns

Index([u'DRG Definition', u'Provider Id', u'Provider Name',
       u'Provider Street Address', u'Provider Zip Code',
       u'Hospital Referral Region Description', u'Total Discharges',
       u'Average Covered Charges', u'Average Total Payments',
       u'Average Medicare Payments'],
      dtype='object')

A sample of the dataset:

In [5]:
df.sample(5, random_state=1234).head()

DRG Definition Provider Id  \
30075             193 - SIMPLE PNEUMONIA & PLEURISY W MCC      210033   
53790   280 - ACUTE MYOCARDIAL INFARCTION, DISCHARGED ...      150026   
35917        195 - SIMPLE PNEUMONIA & PLEURISY W/O CC/MCC      260180   
1627    057 - DEGENERATIVE NERVOUS SYSTEM DISORDERS W/...      230024   
143707    690 - KIDNEY & URINARY TRACT INFECTIONS W/O MCC      360112   

                          Provider Name Provider Street Address  \
30075           CARROLL HOSPITAL CENTER        200 MEMORIAL AVE   
53790         IU HEALTH GOSHEN HOSPITAL       200 HIGH PARK AVE   
35917      CHRISTIAN HOSPITAL NORTHEAST         11133 DUNN ROAD   
1627               SINAI-GRACE HOSPITAL      6071 W OUTER DRIVE   
143707  MERCY ST VINCENT MEDICAL CENTER      2213 CHERRY STREET   

       Provider Zip Code Hospital Referral Region Description  \
30075              21157                       MD - Baltimore   
53790              46526                      IN - South Bend   
35917              63136                       MO - St. Louis   
1627               48235                         MI - Detroit   
143707             43608                          OH - Toledo   

       Total Discharges Average Covered Charges Average Total Payments  \
30075                23             $16,013.95             $15,102.30    
53790                11             $34,427.00              $9,429.09    
35917                26             $16,288.38              $4,093.92    
1627                 22             $14,041.63              $8,260.90    
143707               80             $20,527.33              $6,425.51    

       Average Medicare Payments  
30075                $13,555.52   
53790                 $8,711.63   
35917                 $3,312.69   
1627                  $7,392.00   
143707                $5,713.80

## Q1. Transform relevant columns

All columns in the dataset are of type str. At least 4 of them should be numeric.

In [6]:
df.head()

DRG Definition Provider Id  \
0  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10001   
1  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10005   
2  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10006   
3  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10011   
4  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10016   

                      Provider Name     Provider Street Address  \
0  SOUTHEAST ALABAMA MEDICAL CENTER      1108 ROSS CLARK CIRCLE   
1     MARSHALL MEDICAL CENTER SOUTH  2505 U S HIGHWAY 431 NORTH   
2    ELIZA COFFEE MEMORIAL HOSPITAL          205 MARENGO STREET   
3                 ST VINCENT'S EAST  50 MEDICAL PARK EAST DRIVE   
4     SHELBY BAPTIST MEDICAL CENTER     1000 FIRST STREET NORTH   

  Provider Zip Code Hospital Referral Region Description Total Discharges  \
0             36301                          AL - Dothan               91   
1             35957                      AL - Birmingham               14   
2             35631                      AL - Birmingham               24   
3             35235                      AL - Birmingham               25   
4             35007                      AL - Birmingham               18   

  Average Covered Charges Average Total Payments Average Medicare Payments  
0             $32,963.07              $5,777.24                 $4,763.73   
1             $15,131.85              $5,787.57                 $4,976.71   
2             $37,560.37              $5,434.95                 $4,453.79   
3             $13,998.28              $5,417.56                 $4,129.16   
4             $31,633.27              $5,658.33                 $4,851.44

1.1 Identify them and convert the values of these columns to numeric.

In [7]:
df['Total Discharges'] = pd.to_numeric(df['Total Discharges'])

In [8]:
df['Average Covered Charges']= pd.to_numeric(df['Average Covered Charges'].apply(
                                                lambda x: x[1:].replace(',','')))

In [9]:
df['Average Total Payments']=pd.to_numeric(df['Average Total Payments'].apply(
                                                lambda x: x[1:].replace(',','')))

In [10]:
df['Average Medicare Payments']= pd.to_numeric(df['Average Medicare Payments'].apply(
                                                lambda x: x[1:].replace(',','')))

1.2 Explain why these values should be numeric.

We will need these data in numeric instead of string to compare the value.

## Q2. Let's take a look at heart disease

**Be careful when you identify 'heart' disease.** There are more than one type of DRG definition related to heart diseases. Make sure you catch all of them.  (hint 1:  use .unique() to start your investigation. hint 2: there are two keywords can be used to find all 'heart' diseases.

2.1, Total how many heart related disease cases across the whole country ?

In [11]:
df_cardi=df[(df['DRG Definition'].str.contains('HEART')) | 
            (df['DRG Definition'].str.contains('CARDI'))]

In [12]:
df_cardi.head(1)

DRG Definition Provider Id  \
39739  238 - MAJOR CARDIOVASC PROCEDURES W/O MCC       70033   

          Provider Name Provider Street Address Provider Zip Code  \
39739  DANBURY HOSPITAL         24 HOSPITAL AVE              6810   

      Hospital Referral Region Description  Total Discharges  \
39739                       CT - New Haven                32   

       Average Covered Charges  Average Total Payments  \
39739                 70117.68                25146.03   

       Average Medicare Payments  
39739                   22800.37

In [13]:
df_cardi['Total Discharges'].sum()

1189090

2.2, Create an overview from the view point of 'Hospital Referral Region Description' and 'Average Total Payments', sorted by payments in descending order. List the top 10 only.

In [14]:
df_cardi.groupby('Hospital Referral Region Description')['Average Total Payments']\
                .describe().sort_values('mean',ascending=False).head(10)

count          mean          std  \
Hospital Referral Region Description                                     
CA - San Mateo County                  47.0  16805.429574  9979.927368   
NY - Manhattan                        279.0  16207.337384  7877.421900   
CA - Salinas                           32.0  15910.470312  9908.305755   
CA - San Jose                          75.0  15351.376400  8224.114956   
NY - Bronx                             64.0  15077.005312  7202.381527   
CA - San Francisco                     89.0  14535.558202  7672.439022   
CA - Alameda County                    80.0  14445.442625  7544.800209   
MD - Baltimore                        251.0  14117.053108  9009.682569   
AK - Anchorage                         40.0  14001.218500  7088.255041   
CA - Santa Rosa                        35.0  13663.912857  7185.018671   

                                          min        25%        50%  \
Hospital Referral Region Description                                  
CA - San Mateo County                 4407.00  8558.2600  14226.510   
NY - Manhattan                        4743.69  9757.7050  14895.420   
CA - Salinas                          4908.72  8522.7950  14647.025   
CA - San Jose                         4585.00  8698.5400  14219.480   
NY - Bronx                            4514.20  9726.7675  13455.025   
CA - San Francisco                    4839.00  8328.3800  13383.150   
CA - Alameda County                   5157.63  8837.8925  12814.805   
MD - Baltimore                        3854.75  7402.1400  11453.460   
AK - Anchorage                        5160.63  8313.2300  11527.630   
CA - Santa Rosa                       4761.72  8400.0350  10894.680   

                                             75%       max  
Hospital Referral Region Description                        
CA - San Mateo County                 23083.0050  40864.44  
NY - Manhattan                        20942.7950  41904.15  
CA - Salinas                          19468.4800  44286.90  
CA - San Jose                         20117.7600  38362.13  
NY - Bronx                            19870.6225  37446.43  
CA - San Francisco                    17018.2500  42422.60  
CA - Alameda County                   18692.8525  36598.07  
MD - Baltimore                        18121.0700  54682.48  
AK - Anchorage                        18414.1575  33012.57  
CA - Santa Rosa                       17830.5050  33133.73

2.3, Find out which hospital treat heart disease in "CA - Alameda County" region, sorted by the total case number they handle by descending order.

In [68]:
df_Alameda = df_cardi[df_cardi['Hospital Referral Region Description'].\
    str.contains('CA - Alameda County')]

In [69]:
len(df_Alameda)

80

In [71]:
df_Alameda.groupby('Provider Name').\
            agg({'Total Discharges':'sum'}).sort_values(by = \
                    'Total Discharges',ascending = False)

Total Discharges
Provider Name                                                       
ALTA BATES SUMMIT MEDICAL CENTER                                 658
WASHINGTON HOSPITAL                                              430
EDEN MEDICAL CENTER                                              400
ALTA BATES SUMMIT MEDICAL CENTER - ALTA BATES CAMP               297
ST ROSE HOSPITAL                                                 264
VALLEYCARE MEDICAL CENTER                                        197
ALAMEDA HOSPITAL                                                 118
KAISER FOUNDATION HOSPITAL OAKLAND/RICHMOND                       70
ALAMEDA COUNTY MEDICAL CENTER                                     62
SAN LEANDRO HOSPITAL                                              33
KAISER FOUNDATION HOSPITAL FREMONT/HAYWARD                        27

## Q3
Merge the data set with mapping of US Zipcode, county, and state information from Census data.
Read the file `geo-data.csv` into an object `zip_codes`(make sure that you read all columns as string). After merged, there may have some columns (state and state_abbr) have nan values, fill them in with the information you can find in the first dataframe.

3.1 Based on the merged dataframe, find out the total DRG number for each city in California's Santa Clara county.

In [16]:
zip_codes = pd.read_csv("geo-data.csv",dtype=str)

In [17]:
zip_codes.rename(columns = {'zipcode':'Provider Zip Code'}, inplace=True)

In [18]:
zip_codes.head(1)

state_fips    state state_abbr Provider Zip Code     county   city
0          1  Alabama         AL             35004  St. Clair  Acmar

In [19]:
df_merged=df.merge(zip_codes, how='left', left_on='Provider Zip Code', \
                                          right_on='Provider Zip Code')

the zip_codes data set is not clean. Has some duplications.

In [20]:
zip_codes[zip_codes['Provider Zip Code']=='52761']

state_fips     state state_abbr Provider Zip Code       county       city
6624         17  Illinois         IL             52761  Rock Island        NaN
9728         19      Iowa         IA             52761    Muscatine  Muscatine

In [21]:
len(df)

163065

In [22]:
len(df_merged)

163166

In [23]:
df_merged.head()

DRG Definition Provider Id  \
0  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10001   
1  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10005   
2  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10006   
3  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10011   
4  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10016   

                      Provider Name     Provider Street Address  \
0  SOUTHEAST ALABAMA MEDICAL CENTER      1108 ROSS CLARK CIRCLE   
1     MARSHALL MEDICAL CENTER SOUTH  2505 U S HIGHWAY 431 NORTH   
2    ELIZA COFFEE MEMORIAL HOSPITAL          205 MARENGO STREET   
3                 ST VINCENT'S EAST  50 MEDICAL PARK EAST DRIVE   
4     SHELBY BAPTIST MEDICAL CENTER     1000 FIRST STREET NORTH   

  Provider Zip Code Hospital Referral Region Description  Total Discharges  \
0             36301                          AL - Dothan                91   
1             35957                      AL - Birmingham                14   
2             35631                      AL - Birmingham                24   
3             35235                      AL - Birmingham                25   
4             35007                      AL - Birmingham                18   

   Average Covered Charges  Average Total Payments  Average Medicare Payments  \
0                 32963.07                 5777.24                    4763.73   
1                 15131.85                 5787.57                    4976.71   
2                 37560.37                 5434.95                    4453.79   
3                 13998.28                 5417.56                    4129.16   
4                 31633.27                 5658.33                    4851.44   

  state_fips    state state_abbr     county          city  
0          1  Alabama         AL    Houston        Taylor  
1          1  Alabama         AL   Marshall          Boaz  
2        NaN      NaN        NaN        NaN           NaN  
3          1  Alabama         AL  Jefferson  Center point  
4          1  Alabama         AL     Shelby      Keystone

In [24]:
len(df_merged[df_merged.state.isnull()])

14262

In [25]:
len(df_merged[df_merged.state_abbr.isnull()])

14262

In [26]:
len(df_merged[df_merged.county.isnull()])

14262

In [27]:
len(df_merged[df_merged.city.isnull()])

14363

Get the state_abbr from the column 'Hospital Referral Region Description' letters in front of the '-'. Use a lambda function to split the characters "AL - Dothan", then take the first part.

In [28]:
df_merged['Hospital Referral Region Description'].apply(lambda x: x.split(' - ')[0])[0:5]

0    AL
1    AL
2    AL
3    AL
4    AL
Name: Hospital Referral Region Description, dtype: object

In [29]:
df_merged.loc[df_merged.state.isnull(), 'state_abbr'] = df_merged[
    'Hospital Referral Region Description'].apply(lambda x : x.split(" - ")[0])

Wait for few seconds, then check whether we still have 'state_abbr' is null 

In [30]:
len(df_merged[df_merged.state_abbr.isnull()])

0

Before fill in the missing 'state' , we can answer the question already.

In [31]:
df_SantaClara=df_merged[df_merged.county=='Santa Clara']

In [32]:
df_SantaClara.groupby('city')['DRG Definition'].count()

city
Gilroy            31
Mountain view     81
San jose         285
Santa clara        8
Stanford          87
Name: DRG Definition, dtype: int64

Now we have the state_abbr from 1st DataFrame, need to refer to original zip_codes based on state_abbr then find the state name.

In [33]:
zip_codes_temp = zip_codes[['state','state_abbr']]

In [34]:
d = dict(zip(zip_codes_temp.state_abbr, zip_codes_temp.state))

In [35]:
d

{'AK': 'Alaska',
 'AL': 'Alabama',
 'AR': 'Arkansas',
 'AZ': 'Arizona',
 'CA': 'California',
 'CO': 'Colorado',
 'CT': 'Connecticut',
 'DC': 'District of columbia',
 'DE': 'Delaware',
 'FL': 'Florida',
 'GA': 'Georgia',
 'HI': 'Hawaii',
 'IA': 'Iowa',
 'ID': 'Idaho',
 'IL': 'Illinois',
 'IN': 'Indiana',
 'KS': 'Kansas',
 'KY': 'Kentucky',
 'LA': 'Louisiana',
 'MA': 'Massachusetts',
 'MD': 'Maryland',
 'ME': 'Maine',
 'MI': 'Michigan',
 'MN': 'Minnesota',
 'MO': 'Missouri',
 'MS': 'Mississippi',
 'MT': 'Montana',
 'NC': 'North carolina',
 'ND': 'North dakota',
 'NE': 'Nebraska',
 'NH': 'New hampshire',
 'NJ': 'New jersey',
 'NM': 'New mexico',
 'NV': 'Nevada',
 'NY': 'New york',
 'OH': 'Ohio',
 'OK': 'Oklahoma',
 'OR': 'Oregon',
 'PA': 'Pennsylvania',
 'RI': 'Rhode island',
 'SC': 'South carolina',
 'SD': 'South dakota',
 'TN': 'Tennessee',
 'TX': 'Texas',
 'UT': 'Utah',
 'VA': 'Virginia',
 'VT': 'Vermont',
 'WA': 'Washington',
 'WI': 'Wisconsin',
 'WV': 'West virginia',
 'WY': 'Wyoming

Or ....  

d = zip_codes_temp.set_index('state_abbr').T.to_dict('list')

df_merged.loc[df_merged.state.isnull(), 'state']= df_merged['state_abbr'].apply(lambda x: d[x][0])

check how many 'state' are still null

In [36]:
len(df_merged[df_merged.state.isnull()])

14262

In [37]:
df_merged.loc[df_merged.state.isnull(), 'state']= df_merged['state_abbr'].apply(lambda x: d[x])

check how many 'state' are still null

In [55]:
len(df_merged[df_merged.state.isnull()])

0

In [57]:
df_merged.head()

DRG Definition Provider Id  \
0  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10001   
1  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10005   
2  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10006   
3  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10011   
4  039 - EXTRACRANIAL PROCEDURES W/O CC/MCC       10016   

                      Provider Name     Provider Street Address  \
0  SOUTHEAST ALABAMA MEDICAL CENTER      1108 ROSS CLARK CIRCLE   
1     MARSHALL MEDICAL CENTER SOUTH  2505 U S HIGHWAY 431 NORTH   
2    ELIZA COFFEE MEMORIAL HOSPITAL          205 MARENGO STREET   
3                 ST VINCENT'S EAST  50 MEDICAL PARK EAST DRIVE   
4     SHELBY BAPTIST MEDICAL CENTER     1000 FIRST STREET NORTH   

  Provider Zip Code Hospital Referral Region Description  Total Discharges  \
0             36301                          AL - Dothan                91   
1             35957                      AL - Birmingham                14   
2             35631                      AL - Birmingham                24   
3             35235                      AL - Birmingham                25   
4             35007                      AL - Birmingham                18   

   Average Covered Charges  Average Total Payments  Average Medicare Payments  \
0                 32963.07                 5777.24                    4763.73   
1                 15131.85                 5787.57                    4976.71   
2                 37560.37                 5434.95                    4453.79   
3                 13998.28                 5417.56                    4129.16   
4                 31633.27                 5658.33                    4851.44   

  state_fips    state state_abbr     county          city  
0          1  Alabama         AL    Houston        Taylor  
1          1  Alabama         AL   Marshall          Boaz  
2        NaN  Alabama         AL        NaN           NaN  
3          1  Alabama         AL  Jefferson  Center point  
4          1  Alabama         AL     Shelby      Keystone

## Q4
4.1 Identify five hospitals and show the hospital name, city and state information that offer lowest spending on lung disease treatment 

hint 1: there are two keywords can be used to find all 'lung' diseases. 

hint 2: do not include any RESPIRATORY SYSTEM diseases.

In [39]:
df_lung = df_merged[(df_merged['DRG Definition'].str.contains('PULMONARY')) | (
        df_merged['DRG Definition'].str.contains('PNEUMONIA'))]

#### If use 'Average Total Payments'

In [72]:
df_lung.groupby(['Provider Name','city','state'])[
                'Average Total Payments'].mean().sort_values().head()

Provider Name                 city           state   
BULLOCK COUNTY HOSPITAL       Union springs  Alabama     4005.230000
GROVE HILL MEMORIAL HOSPITAL  Grove hill     Alabama     4114.536667
SARTORI MEMORIAL HOSPITAL     Cedar falls    Iowa        4309.490000
JACKSON MEDICAL CENTER        Jackson        Alabama     4320.563333
ST JOHN OWASSO                Owasso         Oklahoma    4321.625000
Name: Average Total Payments, dtype: float64

#### If use 'Average Covered Payments'

In [40]:
df_lung.groupby(['Provider Name','city','state'])[
                'Average Covered Charges'].mean().sort_values().head()

Provider Name                 city             state      
LAKE WHITNEY MEDICAL CENTER   Bonanza          Texas          4309.226667
YALOBUSHA GENERAL HOSPITAL    Water valley     Mississippi    4589.325000
J PAUL JONES HOSPITAL         Camden           Alabama        4616.580000
EAST CARROLL PARISH HOSPITAL  Lake providence  Louisiana      4765.640000
KILMICHAEL HOSPITAL           Kilmichael       Mississippi    4896.420000
Name: Average Covered Charges, dtype: float64

## Q5

For a patient who has "misc disorders of nutrition,metabolism,fluids/electrolytes", which hospital across the whole United States will be able to offer the lowest cost from average total payments point of view.

5.1 Please find that hospital.

In [41]:
P_sympton = "misc disorders of nutrition,metabolism,fluids/electrolytes".upper()


In [42]:
df_P_sympton = df_merged[df_merged['DRG Definition'].str.contains(P_sympton)]


In [43]:
df_P_sympton[df_P_sympton['Average Total Payments'] ==
             df_P_sympton['Average Total Payments'].min()]

DRG Definition Provider Id  \
134051  641 - MISC DISORDERS OF NUTRITION,METABOLISM,F...      510023   

                 Provider Name Provider Street Address Provider Zip Code  \
134051  WEIRTON MEDICAL CENTER        601 COLLIERS WAY             26062   

       Hospital Referral Region Description  Total Discharges  \
134051                      PA - Pittsburgh                27   

        Average Covered Charges  Average Total Payments  \
134051                  6899.44                 3286.11   

        Average Medicare Payments state_fips          state state_abbr  \
134051                    2409.22         54  West virginia         WV   

         county     city  
134051  Hancock  Weirton

In [44]:
df_P_sympton[df_P_sympton['Average Total Payments'] ==
             df_P_sympton['Average Total Payments'].min()]['Provider Name'].values[0]

'WEIRTON MEDICAL CENTER'

5.2 From 'Average Covered Charges' point of view, What's the ratio between the city with the highest average and the lowest average ?
What are those two cities ?

In [45]:
nlargest_P = df_P_sympton.groupby('city')['Average Covered Charges'].mean().nlargest(1)

In [46]:
nsmallest_P = df_P_sympton.groupby('city')['Average Covered Charges'].mean().nsmallest(1)

In [47]:
nlargest_P.values[0]/nsmallest_P.values[0]

33.969732375042149

In [48]:
nlargest_P.index[0]

'Stanford'

In [49]:
nsmallest_P.index[0]

'Dilley'

## Q6 

6.1 For California, which disease is the most costly one?

In [50]:
df_merged[df_merged.state == "California"].groupby('DRG Definition')['Average Covered Charges'].mean().sort_values(ascending=False).head(1)

DRG Definition
870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS    277004.642419
Name: Average Covered Charges, dtype: float64

6.2 For each state, find the disease that is the most costly one

In [54]:
expensive = df_merged.groupby(['state', 'DRG Definition'])[['Average Covered Charges']].mean()
expensive.apply(lambda x : x.sort_values(ascending=False)).groupby(level=0).head(1)

Average Covered Charges
state                DRG Definition                                                             
California           870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              277004.642419
Nevada               853 - INFECTIOUS & PARASITIC DISEASES W O.R. PR...            257844.555000
New jersey           870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              247313.996667
Colorado             207 - RESPIRATORY SYSTEM DIAGNOSIS W VENTILATOR...            218676.036000
Pennsylvania         870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              190952.952105
Alaska               870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              190709.900000
Florida              870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              181357.381429
Texas                870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              180563.502535
Kansas               870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              176302.070000
District of columbia 853 - INFECTIOUS & PARASITIC DISEASES W O.R. PR...            172520.977500
South dakota         870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              172405.330000
Arizona              870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              170820.167500
Oklahoma             870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              167452.327273
Nebraska             853 - INFECTIOUS & PARASITIC DISEASES W O.R. PR...            163205.201667
Alabama              207 - RESPIRATORY SYSTEM DIAGNOSIS W VENTILATOR...            163033.033810
Minnesota            870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              158667.800000
South carolina       870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              157255.484000
Missouri             870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              156365.340435
Washington           870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              154381.747692
Oregon               870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              153319.652500
Rhode island         870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              146844.600000
Mississippi          329 - MAJOR SMALL & LARGE BOWEL PROCEDURES W MCC              141065.992778
Louisiana            853 - INFECTIOUS & PARASITIC DISEASES W O.R. PR...            140171.755200
Hawaii               329 - MAJOR SMALL & LARGE BOWEL PROCEDURES W MCC              138147.285000
Georgia              870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              137265.971562
Illinois             870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              136386.289200
Tennessee            870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              135035.465000
Connecticut          207 - RESPIRATORY SYSTEM DIAGNOSIS W VENTILATOR...            133246.514444
New mexico           207 - RESPIRATORY SYSTEM DIAGNOSIS W VENTILATOR...            133136.320000
New hampshire        207 - RESPIRATORY SYSTEM DIAGNOSIS W VENTILATOR...            132782.463333
Wyoming              853 - INFECTIOUS & PARASITIC DISEASES W O.R. PR...            131640.840000
Virginia             870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              127637.295000
Wisconsin            207 - RESPIRATORY SYSTEM DIAGNOSIS W VENTILATOR...            126561.048889
Indiana              870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              122649.482941
New york             870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              121972.016154
Ohio                 870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              121413.748049
Iowa                 207 - RESPIRATORY SYSTEM DIAGNOSIS W VENTILATOR...            120564.112500
Idaho                870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              117492.085000
Kentucky             870 - SEPTICEMIA OR SEVERE SEPSIS W MV 96+ HOURS              115098.180000
Arkansas             329 - MAJOR SMALL & LARGE BOWEL PROCEDURES W MCC              109527.821000

6.3 Based on the above results, provide your analysis.

DRG 870 is the most costly diseases in US. US goverment or health agencies  should do something in public education to reduce the overall cost in the future. 

## Q7
Based on the merged data set, describe the most interesting or unexpected finding you may have. (note: you need to list the related code to back up/prove your statement.

This is something for you to explore/analysis ...